json-smart/src/main/java/net/minidev/json/parser/JSONParserBase.java

```diff
@@ -141,9 +141,13 @@ protected Number extractFloat() throws ParseException {
 			checkLeadinZero();
 		if (!useHiPrecisionFloat)
 			return Float.parseFloat(xs);
-		if (xs.length() > 18) // follow JSonIJ parsing method
-			return new BigDecimal(xs);
-		return Double.parseDouble(xs);
+		double value = Double.parseDouble(xs);
+		if (xs.length() > 18) { // follow JSonIJ parsing method
+			if(!String.valueOf(value).equals(xs)) {
```

Comment:
like that that will rune performances ...
it's obvious that a number with 50 number will be a big decimal.

Reply:
There is a problem here, it just compare the string equlas, but ignore the case of `e` `E` `e+` `E+`